src/bejeweled/game/GameLogic.java

```diff
@@ -117,23 +118,55 @@ public void returnFromAnimation() {
 	 * untill there are no more combinations then the animations will end
 	 */
 	private void checkForCombinations() {
-		ArrayList<Gem> combinations = board.checkForCombinations();
-		for(Gem g : combinations) {
-			if(g instanceof DeleteRowGem) {
-				combinations = board.deleteRowAndCol(g, combinations);
+		ArrayList<Combination> combinations = board.checkForCombinations();
+		boolean br = false;
+		for (Combination b : combinations) {
+			br = false;
+			for (Gem g : b.getGems()) {
+				if (g instanceof DeleteRowGem) {
+					Combination combi = board.deleteRowAndCol(g, combinations);
+					combinations.add(combi);
+					br = true;
+				}
+			}
+			if (br) {
 				break;
 			}
 		}
-		//combinations found
+		while(br) {
+			br = false;
+			for(Gem g : combinations.get(combinations.size()-1).getGems()) {
+				if (g instanceof DeleteRowGem) {
+					Combination combi = board.deleteRowAndCol(g, combinations);
+					combinations.add(combi);
+					br = true;
+				}
+				if(br) {
+					break;
+				}
+			}
+			
+		}
+		
+		// combinations found
 		if(combinations.size()>0){
+			int gems = 0;
+			for(int i = 0; i < combinations.size(); i++) {
+				gems += combinations.get(i).getSize();
+			}
 			combinationsFormed = true;
 			Sounds.getInstance().playCombinationSound();
-			score.updateScore(combinations.size()); //update score
+			int timesScore = 1;
 			for(int i = 0; i < combinations.size(); i++) {
-				if(combinations.get(i) instanceof DoublePointsGem) {
-					score.updateScore(combinations.size());
+				for(Gem gem : combinations.get(i).getGems()) {
+					if(gem instanceof DoublePointsGem) {
+						timesScore *= 2;
```

Comment:
That doesn't double the score. When there are 2 doublePointGems, you want the score*2*2 which means the score*4. If you would place it in the earlier loop the result would've been the score*3, so that's why I implemented it this way.